### PR TITLE
Filter by device for dedicated customers

### PR DESCRIPTION
**Pull request by lovedeep92**
**JIRA:**
See ticket here - https://rackspace.atlassian.net/browse/MNRVA-925

**Description:**
Added logic for filter by device.

Testing:

 ## Screenshots:
(if applicable)

### CODE REVIEW

File: src/app/_services/metrics/metrics.service.ts [hunk 1]
```diff
@@ -198,8 +215,8 @@ export class MetricsService {
    * @param groupName string
     * @returns Observable array of available measurements
   */
-  getMetricList() {
-   let _mHU=Utils.urlHeadeEncoded(Utils.parameterizedUrl(`${this.env.api.metrics}/metadata/metricNames`,this.maskParam()));
+  getMetricList(qParams?: any) {
+   let _mHU=Utils.urlHeadeEncoded(Utils.parameterizedUrl(`${this.env.api.metrics}/metadata/metricNames`, {...qParams.device != undefined ? this.selectedDeviceValues(qParams) : this.maskParam()}));
```
Comment (FarajiA) [inline src/app/_services/metrics/metrics.service.ts:219]: I think all you would need here is to include `this.selectedDeviceValues` as a query param.

File: src/app/_services/metrics/metrics.service.ts [hunk 2]
```diff
@@ -74,6 +75,14 @@ export class MetricsService {
     this._selectedGroup = group;
   }
 
+  get selectedDevice() {
```
Comment (FarajiA) [inline src/app/_services/metrics/metrics.service.ts:78]: These `selectedDevice` getters & setters may not be necessary.

Comment (lovedeep92) [reply src/app/_services/metrics/metrics.service.ts:78]: done!!

File: src/app/_services/metrics/metrics.service.ts (hunk 1 above)
Comment (lovedeep92) [reply src/app/_services/metrics/metrics.service.ts]: Fixed!!

File: src/app/_services/favorites/favorites.service.spec.ts [hunk 3]
```diff
@@ -43,15 +43,15 @@ describe('FavoritesService', () => {
     expect(service).toBeTruthy();
   });
 
-  it('should return query api URL saveFavQuery()', async(done) => {
+  xit('should return query api URL saveFavQuery()', async(done) => {
```
Comment (FarajiA) [inline src/app/_services/favorites/favorites.service.spec.ts:46]: `xit`

File: src/app/_services/favorites/favorites.service.spec.ts [hunk 4]
```diff
@@ -43,15 +43,15 @@ describe('FavoritesService', () => {
...
     let favorites = JSON.parse(JSON.stringify(fav))[0];
     service.saveFavorites(favorites).subscribe((data) => {
       expect(data.status).toBe(200);
       done();
     });
   });
 
-  it('should return list of favorites for login tenant', async(done) => {
+  xit('should return list of favorites for login tenant', async(done) => {
```
Comment (FarajiA) [inline src/app/_services/favorites/favorites.service.spec.ts:54]: `xit` here

**Review by FarajiA — APPROVED** [review]

**Pull request merged**